Add GaussianDropout

the-moliver opened the pull request:
Now that the noise module is official I wanted to push my original noise layer addition of Gaussian Dropout, which is multiplicative (rather than additive) Gaussian Noise.

fchollet commented on keras/layers/noise.py:
```diff
@@ -20,4 +20,26 @@ def get_output(self, train=False):
+        X = self.get_input(train)
+        if train:
+            # self.p refers to drop probability rather than retain probability (as in paper) to match Dropout layer syntax
+            X *= srng.normal(size=X.shape, avg = 1., std=T.sqrt(self.p/(1-self.p)), dtype=theano.config.floatX)
```
> `T.sqrt(self.p/(1-self.p))`

Something should be done to avoid division by zero in case p = 1... maybe clipping p to [epsilon, 1-epsilon]?

In terms of coding style, please use spaces around operators and use floats in float operations.

the-moliver commented:
Added the spaces and changed 1 to 1.0. This is meant to be a drop-in replacement for a dropout layer as in the paper, i.e. `Dropout(.5)` could be changed to `GaussianDropout(.5)`. So in either case `p=1.0` doesn't really make sense and would break things.

simonhughes22 commented:
Can you update the module with a link to the appropriate paper(s)?

the-moliver commented:
That is the appropriate paper! Read it! ;-)

simonhughes22 commented:
Sorry I was referring to the regular GaussianLayer. Also, a url would be good.

the-moliver commented:
URL added for GaussianDropout

simonhughes22 commented:
Awesome, thanks.
